### ISSUE TITLE
Avoid generating code with a redundant (Object) cast

### ISSUE THREAD

**Issue by Stephan202** [jOOQ/jOOQ#6705]
We build our code with `-Werror`. Unfortunately the generated code contains one section that triggers a warning:
```java
    /**  
     * @deprecated Unknown data type. Please define an explicit {@link org.jooq.Binding} to specify how this type should be handled.
     */
    @java.lang.Deprecated
    public Object getWay() {
        return (Object) get(68);
    }
```

For the code, `javac` emits the following warning:
```
redundant cast to java.lang.Object
```

It'd be nice if the generated code does not perform this cast. Will open a PR with a proposed fix shortly.

(Test against jOOQ 3.9.3 as well as the code currently on `master`. Java 8, PostgreSQL.)

**Comment by lukaseder** [comment]
Thank you very much for your report and for your suggested fix. I completely agree with the optimisation.

**Comment by lukaseder** [comment]
This will be merged also to 3.10.2 (#6711)
